### ISSUE TITLE
BC: if key is set to token, the new tokenKey default will overwrite it

### ISSUE THREAD

**Issue by danwkennedy** [koajs/jwt#77]
I realize this is very nit-picky.

Commit 9cf6a15c03948fd7b415bfa8d0c7946020b6100b adds a tokenKey that writes the raw token to the key specified. This'll break backwards compatibility if the "key" preference is set to 'token' like this:
```
const jwt = require('koa-jwt');
jwt({ key: 'token'});
```

In that case, the value at token will be the user variable in versions <=1.2.0 but the raw string token in 1.3.0 >=.

**Comment by sc0ttyd** [comment]
Hi @danwkennedy - you're right, the implementation in v2.x protected against this but it needs backporting to v1.x. I'll fix it.

No, I don't think that is too nit-picky! Thanks

**Comment by sc0ttyd** [comment]
https://github.com/koajs/jwt/blob/master/index.js#L50 fixes this in 1.3.0. I'll backport the fix to 1.2.x

**Comment by sc0ttyd** [comment]
Hi @danwkennedy, although the issue in question was introduced in a push to GitHub in https://github.com/koajs/jwt/commit/9cf6a15c03948fd7b415bfa8d0c7946020b6100b, the repo was not published to npm until 1.3.0, when this issue was fixed. As such, there are no versions released in the wild that suffer from this issue - it only occurs when someone checks out the repo at a commit between versions.

**Comment by danwkennedy** [comment]
The check here https://github.com/koajs/jwt/blob/master/index.js#L50 will always resolve to true since the value is defaulted to 'token' here https://github.com/koajs/jwt/blob/master/index.js#L12 so the decoded token will get set to the key 'token' and then immediately afterwards, the encoded token will get set to the key 'token'.

This currently breaks our implementation when going from 1.2.0 to 1.3.0. The solution is to switch the configuration from:
```
const jwt = require('koa-jwt');
jwt({ key: 'token'});
```
to
```
const jwt = require('koa-jwt');
jwt({ key: 'token', tokenKey: 'rawToken'});
```

**Comment by sc0ttyd** [comment]
@danwkennedy my bad, apologies. This should fix it, could you review please? https://github.com/koajs/jwt/pull/78

**Comment by danwkennedy** [comment]
Looks good. I'm a little apprehensive of taking out the default, though. This is literally for one potential value of key. Perhaps when v2 comes out, it would be good to add it back?

Thanks for responding to the issue so quickly!

**Comment by sc0ttyd** [comment]
No problem! `opts.tokenKey` support was only in the wild for a few days so there will be few if any users relying on that functionality. Plus the fix for those users (adding `{ tokenKey: 'token' }`) is trivial, whereas both identifying the problem and fixing it without this PR for users like you, who are expecting `ctx.state.token` not to be touched, is more difficult. I can push this to npm as a patch release later today.

**Comment by sc0ttyd** [comment]
Published as v2.2.1
